krux_troposphere/troposphere.py

```diff
@@ -0,0 +1,169 @@
+        # causing an error. This creates a way to circumvent that.
+
+        # Add all the boto arguments
+        add_boto_cli_arguments(parser)
+
+    # Add those specific to the application
+    group = get_group(parser, NAME)
+
+
+class Troposphere(object):
+    """
+    A manager to handle all Troposphere / Cloud Formation related functions.
+    Each instance is locked to a connection to a designated region (self.boto.cli_region).
+    """
+
+    STACK_NOT_EXIST_ERROR_MSG = 'Stack with id {stack_name} does not exist'
+    NO_UPDATE_ERROR_MSG = 'No updates are to be performed.'
+
+    def __init__(
+        self,
+        boto,
+        logger=None,
+        stats=None,
+    ):
+        """
+        :param boto: :py:class:`krux_boto.boto.Boto3` Boto3 object used to connect to Cloud Formation
+        :param logger: :py:class:`logging.Logger` Logger, recommended to be obtained using krux.cli.Application
+        :param stats: :py:class:`kruxstatsd.StatsClient` Stats, recommended to be obtained using krux.cli.Application
+        """
+        # Private variables, not to be used outside this module
+        self._name = NAME
+        self._logger = logger or get_logger(self._name)
+        self._stats = stats or get_stats(prefix=self._name)
+
+        if not isinstance(boto, Boto3):
+            raise NotImplementedError('Currently krux_troposphere.troposphere.Troposphere only supports krux_boto.boto.Boto3')
+
+        self._cf = boto.client('cloudformation')
+        self.template = troposphere.Template()
+
+    def _is_stack_exists(self, stack_name):
+        """
+        Check if the given Cloud Formation stack exists
+
+        GOTCHA: There is no simple way to check the existence of a stack.
+        The template for the stack is fetched and if an expected exception occur (Unable to find stack),
+        then the stack is deemed not existing.
+
+        :param stack_name: :py:class:`str` Name of the stack to check
+        """
+        try:
+            # See if we can get a template for this
+            self._cf.get_template(StackName=stack_name)
+            # The template was successfully retrieved; the stack exists
+            return True
+        except botocore.exceptions.ClientError as err:
+            if self.STACK_NOT_EXIST_ERROR_MSG.format(stack_name=stack_name) == err.response.get('Error', {}).get('Message', ''):
+                # The template was not retrieved; the stack does not exists
+                return False
+
+            # Unknown error. Raise again.
+            raise
+
+    def save(self, stack_name):
+        """
+        Saves the template to the given Cloud Formation stack.
+
+        The method internally checks whether the stack exists and either creates or updates the stack
+        with the template in this object.
+
+        :param stack_name: :py:class:`str` Name of the stack to check
+        """
+        if self._is_stack_exists(stack_name):
+            try:
+                self._cf.update_stack(
```

Comment:
We probably wont hit the limit, but know that cloud formation has a template size limit for this method. The other option is to save to a temporary s3 location and provide an s3 url in the update request.